File: problems/2016-maximum-difference-between-increasing-elements/solution.cpp
class Solution {
public:
    int maximumDifference(vector<int>& nums) {
        int maxDiff = -1;
        int minVal = nums[0];
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] > minVal) {
                maxDiff = max(maxDiff, nums[i] - minVal);
            } else {
                minVal = nums[i];
            }
        }
        return maxDiff;
    }
};

File: problems/2016-maximum-difference-between-increasing-elements/solution.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        max_diff = -1
        min_val = nums[0]
        for i in range(1, len(nums)):
            if nums[i] > min_val:
                max_diff = max(max_diff, nums[i] - min_val)
            else:
                min_val = min(min_val, nums[i])
        return max_diff